Src/RE2.Managed/Regex2.cs

```diff
@@ -260,16 +260,12 @@ public static unsafe bool Matches(string pattern, string text, out List<Dictiona
                         for (int submatchIndex = 0; submatchIndex < output->NumSubmatches; submatchIndex++)
                         {
                             Submatch submatchRe2 = output->Matches[matchIndex][submatchIndex];
-                            int submatchTextStartIndex = submatchRe2.Index;
-                            int submatchLength = submatchRe2.Length;
 
-                            string submatchString;
-                            if ((submatchTextStartIndex == -1) && (submatchLength == -1))
-                            {
-                                submatchString = string.Empty;
-                            }
-                            else
+                            string submatchString = string.Empty;
+                            if (!submatchRe2.IsOptionalGroupAndNotFound())
                             {
+                                int submatchTextStartIndex = submatchRe2.Index;
+                                int submatchLength = submatchRe2.Length;
                                 submatchString = Encoding.UTF8.GetString(textUtf8Bytes, submatchTextStartIndex, submatchLength);
                             }
```

Comment:
@michaelcfanning This is fewer lines, but it is more confusing to read to me. What do you think?